api/controllers/PromotionController.js

```diff
@@ -41,17 +41,24 @@ var PromotionController = {
 
     getRestaurantById: function(req, res) {
         var id = req.swagger.params.id.value;
+        PromotionController.getRestaurantAndDoSomething(id, PromotionController.returnRestaurantDetail, res, null);
+    },
 
+    getRestaurantAndDoSomething: function(id, callback, res, params) {
```

Comment:
"doSomething" is kinda vague in a method name. think "getRestaurantFromYelp" could be better? also "res", "params" can be omitted since this function does not have to know about them

Reply:
cool. I am changing them.

Reply:
Do you mean that I don't need to define them in the function signature?
Then How can I pass res from createEvent to getRestaurantAndDoSomething and then callback?
I don't quite understand it.

Reply:
Or do you mean by using "The Arguments Object"
I will try it.